H2O-3/introduction-to-machine-learning-with-h2o-3-regression/introduction-to-machine-learning-with-h2o-3-regression.md

```diff
@@ -1,819 +1,1498 @@
+
+search_criteria_tune = {'strategy': "RandomDiscrete",
+                        'max_runtime_secs': 900, #15 min  
+                        'max_models': 100,  ## build no more than 100 models
+                        'seed' : 42}
+
+xgb_grid = H2OGridSearch(xgb, hyper_params,
+                         grid_id = 'random_grid',
+                         search_criteria = search_criteria_tune)
+
+%time xgb_grid.train(x = x, y = y, training_frame = train, validation_frame = valid)
+~~~
+
+Let's print the results of our random search:
+
+~~~python
+sorted_xgb = xgb_grid.get_grid(sort_by = 'rmse', decreasing = False)
+sorted_xgb.sorted_metric_table()
+~~~
+
+Please note that the results might not be exactly the same.
+
+![xgb-top-model](assets/xgb-top-model.png)
+
+With the grid search that we just did, we were able to train a total of 18 models. The score slightly improved from what we obtained in the first grid search. If we were trying to find the best model, we would need to run the grid search for a longer period of time, that way more models can be trained. We could also do a local search with the values we just found and see if we could get better results, but for the purpose of this tutorial, we will stop here.
+
+Let’s retrieve the best model from the grid search and then compare the results from the default model to the tuned model. In order to do so, we also need to save the performance of the best model on the validation set.
+
+Retrieve the best model from the grid search
+~~~python
+best_xgb_model = xgb_grid.models[0] 
+best_xgb_model
+~~~
+Please note that you can also retrieve any of your models by changing the number inside the brackets. Now save the model performance on the validation set.
+
+~~~python
+tuned_xgb_per = best_xgb_model.model_performance(valid)
+~~~
+Print both RMSE and MAE
+
+~~~python
+print("Default XGB RMSE: %.4f \nTuned XGB RMSE:%.4f" % (default_xgb_per.rmse(), tuned_xgb_per.rmse()))
+~~~
+**Output:**
+~~~python
+Default XGB RMSE: 0.4244 
+Tuned XGB RMSE:0.4108
+~~~
+Our RMSE slightly improved with the tuning that we did. As we mentioned before, we only let each random search run for 15 minutes. To see if we can obtain a much better model, we would have to let it run for much longer that way more models can be trained. 
+
+We will check the MAE now:
+
+~~~python
+print("Default XGB MAE: %.4f \nTuned XGB MAE:%.4f" % (default_xgb_per.mae(), tuned_xgb_per.mae()))
+~~~
+**Output:**
+~~~python
+Default XGB MAE: 0.3108 
+Tuned XGB MAE:0.2986
+~~~
+
+In the case of an interest rate decision, an MAE of 0.30 is good enough as it tells us that, on average, the model would predict a very close interest rate to what someone with “x” characteristics should get. However, for some companies that MAE might be too high, so it all depends on the application to decide whether or not our MAE results are satisfactory. We will see if the Deep Learning model can yield a lower RMSE and MAE.
```

Comment:
not sure what you are tryin to say with this sentence: 

"However, for some companies that MAE might be too high, so it all depends on the application to decide whether or not our MAE results are satisfactory."

Consider rewriting the sentence